packages/web-client/tests/unit/layer-1-connection-manager-test.ts

```diff
@@ -0,0 +1,253 @@
+
+    assert.ok(
+      connectionManager.strategy,
+      'There is a strategy after reconnecting'
+    );
+    assert.ok(
+      connectionManager.provider,
+      'There is a provider after reconnecting'
+    );
+    assert.ok(
+      connectionManager.providerId,
+      'There is a providerId after reconnecting'
+    );
+
+    assert.ok(
+      web3.setProvider.calledWith(connectionManager.provider),
+      'web3.setProvider was called with the connection manager provider'
+    );
+  });
+
+  test('Its strategy, provider, and providerId should be cleared after calling reset', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+    let web3 = {
+      setProvider: () => {},
+    };
+    await connectionManager.connect((web3 as unknown) as Web3, 'metamask'); // the test strategy is hardcoded to metamask
+
+    assert.ok(
+      connectionManager.strategy,
+      'There is a strategy after connecting'
+    );
+    assert.ok(
+      connectionManager.provider,
+      'There is a provider after connecting'
+    );
+    assert.ok(
+      connectionManager.providerId,
+      'There is a providerId after connecting'
+    );
+
+    connectionManager.reset();
+
+    assert.ok(!connectionManager.strategy, 'There is no strategy after reset');
+    assert.ok(!connectionManager.provider, 'There is no provider after reset');
+    assert.ok(
+      !connectionManager.providerId,
+      'There is no providerId after reset'
+    );
+  });
+
+  test('It calls the cross-tab-connection callback upon connected messages, if not already connected', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+
+    let onCrossTabConnection = Sinon.spy();
+    connectionManager.on('cross-tab-connection', onCrossTabConnection);
+
+    let message = {
+      type: 'CONNECTED',
+      session: 'SESSION',
+    };
+
+    connectionManager.broadcastChannel.test__simulateMessageEvent(message);
+
+    assert.ok(
+      onCrossTabConnection.calledWith(message),
+      'The cross-tab-connection callback was called with the sent message'
+    );
+  });
+
+  test('It does not call the cross-tab-connection callback upon connected messages, if already connected', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+    let web3 = {
+      setProvider: () => {},
+    };
+
+    let onCrossTabConnection = Sinon.spy();
+    connectionManager.on('cross-tab-connection', onCrossTabConnection);
+
+    await connectionManager.connect((web3 as unknown) as Web3, 'metamask'); // the test strategy is hardcoded to metamask
+
+    let message = {
+      type: 'CONNECTED',
+      session: 'SESSION',
+    };
+
+    connectionManager.broadcastChannel.test__simulateMessageEvent(message);
+
+    assert.ok(
+      onCrossTabConnection.notCalled,
+      'The cross-tab-connection callback was not called'
+    );
+  });
```

Comment:
These two tests are more the kind of thing I think it’s useful to test: that the public interface behaves as expected. 🎉

Reply:
I would like to do more of these tests, however without being able to mock connection behaviour + events in a realistic way for WalletConnect + MetaMask, I think it's hard to do meaningful tests for the other public interfaces (connect, reconnect, disconnect, reset).

Reply:
maybe there’s a way to encapsulate those for mocking, but it’s always hard to know how far/what direction to go etc 🧐